src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThreadGroup.java

```diff
@@ -428,24 +433,47 @@ public void saveProgram(File f) throws IOException {
             return;
         }
 
-        try {
```

Comment:
What's the impact if there is not (another upcoming save operation)? What data gets lost?